atest/testdata/standard_libraries/process/secrets.robot

```diff
@@ -0,0 +1,51 @@
+*** Settings ***
+Suite Setup       Set Environment Variable    v1    system
+Resource          process_resource.robot
+
+*** Variables ***
+@{COMMAND}           python    -c    import os; print(' '.join([os.getenv('v1', '-'), os.getenv('v2', '-'), os.getenv('v3', '-')]))
+${SECRET: Secret}    %{=This is secret!}
+
+*** Test Cases ***
+Run Process with Secret Argument
+    ${result} =    Run Process    python    ${SCRIPT}    ${SECRET}
+    Script result should equal    ${result}    stdout=This is secret!
+
+Run Process with Mixed Arguments Including Secret
+    ${result} =    Run Process    python    ${SCRIPT}    public_arg    ${SECRET}
+    Script result should equal    ${result}    stdout=public_arg    stderr=This is secret!
+
+Run Process with Stdin as Secret
+    ${result} =    Run Process    python    -c    import sys; print(sys.stdin.read())    stdin=${SECRET}    cwd=%{TEMPDIR}
+    Should Be Equal    ${result.stdout}    This is secret!
```

Comment:
Using `Result Should Equal` would make sure also stderr and rc are validated.

Reply:
done, made all changes